src/main/java/com/teamc/fms/controller/TimeSlotController.java

```diff
@@ -0,0 +1,61 @@
+package com.teamc.fms.controller;
+
+
+import com.teamc.fms.dto.CreateTimeSlotDto;
+import com.teamc.fms.dto.TimeSlotDto;
+import com.teamc.fms.entity.TimeSlot;
+import com.teamc.fms.mapper.TimeSlotMapper;
+import com.teamc.fms.model.CreateTimeSlotCommand;
+import com.teamc.fms.service.TimeSlotService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/v1/time-slots")
+public class TimeSlotController {
+    private final TimeSlotService timeSlotService;
+    private final TimeSlotMapper timeSlotMapper;
+
+    @PostMapping("/add")
+    @ResponseStatus(HttpStatus.CREATED)
+    public TimeSlotDto addTimeSlot(@RequestBody  CreateTimeSlotDto createTimeSlotDto) {
+
+        CreateTimeSlotCommand createTimeSlotCommand = timeSlotMapper.toCreateTimeSlotCommand(createTimeSlotDto);
+        TimeSlot timeSlot = timeSlotService.addTimeSlot(createTimeSlotCommand);
+        return timeSlotMapper.toTimeSlotDto(timeSlot);
+    }
+
+    @GetMapping("/all/{facilityId}")
+    @ResponseStatus(HttpStatus.OK)
+    public List<TimeSlotDto> getAllTimeSlots(@PathVariable String facilityId) {
+        List<TimeSlot> timeSlots = timeSlotService.getAllTimeSlots(facilityId);
+        return timeSlots.stream().map(timeSlotMapper::toTimeSlotDto).toList();
+    }
+
+    @GetMapping("/{id}")
+    public TimeSlotDto getTimeSlotById(@PathVariable String id) {
+        TimeSlot timeSlot = timeSlotService.getTimeSlotById(id);
+        return timeSlotMapper.toTimeSlotDto(timeSlot);
+    }
+
+
+@PutMapping("/{id}")
+    public TimeSlotDto updateTimeSlot(@PathVariable("id") String id,
+                                      @RequestBody CreateTimeSlotDto createTimeSlotDto) {
+    CreateTimeSlotCommand createTimeSlotCommand = timeSlotMapper.toCreateTimeSlotCommand(createTimeSlotDto);
+    TimeSlot timeSlot = timeSlotService.updateTimeSlot(id,createTimeSlotCommand);
+    return timeSlotMapper.toTimeSlotDto(timeSlot);
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<String> deleteTimeSlot(@PathVariable("id") String id) {
+        timeSlotService.deleteTimeSlot(id);
+        return ResponseEntity.ok("TimeSlot with ID " + id + " deleted successfully");
+    }
```

Comment:
Please send empty body and 204 NO CONTENT status  -> REST convention
Also, in our convention, we donot return Response entity